plutus-tx/src/Language/PlutusTx/Evaluation.hs

```diff
@@ -52,8 +47,5 @@ evaluateCekTrace
     => Program Name uni fun ()
     -> ([String], CekExTally fun, Either (CekEvaluationException uni fun) (Term Name uni fun ()))
 evaluateCekTrace (Program _ _ t) =
-    let
-        (lg, (res, state)) = unsafePerformIO $ withEmit $ \emit -> do
-            let runtime = toBuiltinsRuntime (DefaultFunDyn emit) PLC.defaultCostModel
-            Control.Exception.evaluate $ runCekCounting runtime t
-    in  (lg, view exBudgetStateTally state, res)
+    case runCek defBuiltinsRuntime Counting True t of
+        (errOrRes, st, logs) -> (logs, view exBudgetStateTally st, errOrRes)
```

Comment:
It kinda sucks that the order of elements in the resulting tuple is different in `plutus-core` and `plutus-tx`. Should we use the same order? Which one?

Reply:
This version looks a bit odd because the result comes last, and you might argue that it's the most important thing.  I think the `plutus-core` version looks more sensible.  Maybe after the budgeting code's been reorganised we'll be able to return exactly what's needed here and dispense with the `view`, so let's make it look as much like the `plutus-core` version as possible in the meantime.

Reply:
I'd be fine with changing `plutus-core` version in this PR, but I don't like the idea of changing `plutus-tx` version as that would be an irrelevant change for this PR. I'll leave it for later then.

Reply:
I have no particular attraction to how this is set up, it's mostly a convenience function.